docs/docs/benchmark.md

```diff
@@ -1,96 +1,262 @@
+
+??? note "Python example"
+    ```python
+    import openml
+    import sklearn
+    benchmark_suite = openml.study.get_study('OpenML-CC18','tasks') # obtain the benchmark suite
+    # build a sklearn classifier
+    clf = sklearn.pipeline.make_pipeline(sklearn.preprocessing.Imputer(),
+                                         sklearn.tree.DecisionTreeClassifier())
+    for task_id in benchmark_suite.tasks: # iterate over all tasks
+      task = openml.tasks.get_task(task_id) # download the OpenML task
+      X, y = task.get_X_and_y() # get the data (not used in this example)
+      openml.config.apikey = 'FILL_IN_OPENML_API_KEY' # set the OpenML Api Key
+      # run classifier on splits (requires API key)
+      run = openml.runs.run_model_on_task(task,clf)
+      score = run.get_metric_score(sklearn.metrics.accuracy_score) # print accuracy score
+      print('Data set: %s; Accuracy: %0.2f' % (task.get_dataset().name,score.mean()))
+      run.publish() # publish the experiment on OpenML (optional)
+      print('URL for run: %s/run/%d' %(openml.config.server,run.run_id))
+    ```
+
+??? note "Java example"
+    ```java
+    public static void runTasksAndUpload() throws Exception {
+      OpenmlConnector openml = new OpenmlConnector();
+      // obtain the benchmark suite
+      Study benchmarksuite = openml.studyGet("OpenML-CC18", "tasks");
+      Classifier tree = new REPTree(); // build a Weka classifier
+      for (Integer taskId : benchmarksuite.getTasks()) { // iterate over all tasks
+        Task t = openml.taskGet(taskId); // download the OpenML task
+        Instances d = InstancesHelper.getDatasetFromTask(openml, t); // obtain the dataset
+        openml.setApiKey("FILL_IN_OPENML_API_KEY");
+        int runId = RunOpenmlJob.executeTask(openml, new WekaConfig(), taskId, tree);
+        Run run = openml.runGet(runId);   // retrieve the uploaded run
+      }
+    }
+    ```
+
+??? note "R example"
+    ```r
+    library(OpenML)
+    lrn = makeLearner('classif.rpart') # construct a simple CART classifier
+    task.ids = getOMLStudy('OpenML-CC18')$tasks$task.id # obtain the list of suggested tasks
+    for (task.id in task.ids) { # iterate over all tasks
+      task = getOMLTask(task.id) # download single OML task
+      data = as.data.frame(task) # obtain raw data set
+      run = runTaskMlr(task, learner = lrn) # run constructed learner
+      setOMLConfig(apikey = 'FILL_IN_OPENML_API_KEY')
+      upload = uploadOMLRun(run) # upload and tag the run
+    }
+    ```
+
+### Retrieving runs on a benchmarking suites:
+Once a benchamrk suite has been created, the listing functions can be used to 
+obtain all results on the benchmark suite. Note that there are several other
```

Comment:
Will there also be a separate call to retrieve all studies for a suite?

Reply:
Good question. This isn't covered yet.